src/hooks/screen/useCreateChat.ts

```diff
@@ -82,19 +128,22 @@ const useCreateChat = () => {
     }
   };
 
-  const handleAnonymousMessage = async (selectedUser) => {
+  const handleAnonymousMessage = async (selectedUser, channelId = null, context = null) => {
     if (!selectedUser?.allow_anon_dm) {
       SimpleToast.show('This user does not allow anonymous messages');
       return;
     }
 
     try {
       const response = await getOrCreateAnonymousChannel(
-        selectedUser?.user_id || selectedUser.userId
+        selectedUser?.user_id || selectedUser.userId,
+        channelId,
+        context
       );
-      const chatData = createChannelJson(response, selectedUser);
+
+      const chatData = createChannelJson(response, selectedUser, true);
       const channelList = ChannelList.fromMessageAnonymouslyAPI(chatData);
-      await channelList.saveIfLatest(localDb);
+      await channelList.save(localDb);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `handleAnonymousMessage` function, you're passing `channelId` and `context` as parameters to `getOrCreateAnonymousChannel`. However, there's no validation or error handling for these parameters. If they are not provided or invalid, this could potentially cause issues. Consider adding appropriate checks or default values.
```diff
- const response = await getOrCreateAnonymousChannel(
-  selectedUser?.user_id || selectedUser.userId,
-  channelId,
-  context
- );
+ const userId = selectedUser?.user_id || selectedUser.userId;
+ if (!userId || !channelId || !context) {
+   throw new Error('Invalid parameters');
+ }
+ const response = await getOrCreateAnonymousChannel(userId, channelId, context);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->